### PR TITLE
fix #295207 : Mscz file corrupted after splitting a measure

### PR DESCRIPTION
**Pull request by njvdberg**
Resolves: https://musescore.org/en/node/295207

The root case was in Score::insertMeasure(). This method always moves time signature, key signatures and clefs to the beginning of the newly created measure. In this was this was not the wanted behavior. This is solved by adding a new boolean, moveSignaturesClef, to the method. If this boolean is false (the default), the method shows the old behavior.
If this boolean is false, the method will not look for time signatures, key signatures or clefs and therefor these are not moved.

Another issue was ties where removed. Also is is corrected.

- [x] I signed [CLA](https://musescore.org/en/cla)
- [x] I made sure the code in the PR follows [the coding rules](https://musescore.org/en/handbook/developers-handbook/finding-your-way-around/musescore-coding-rules)
- [x] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"
- [x] I made sure the commit message title starts with "fix #424242:" if there is a related issue
- [x] I created the test (mtest, vtest, script test) to verify the changes I made

### CODE REVIEW

File: mtest/libmscore/split/tst_split.cpp [hunk 1]
```diff
@@ -42,6 +42,7 @@ class TestSplit : public QObject, public MTest
       void split06() { split("split06.mscx", "split06-ref.mscx", 6); }
       void split07() { split("split07.mscx", "split07-ref.mscx"); }
       void split08() { split("split08.mscx", "split08-ref.mscx"); }
+      void split295207() { split("split295207.mscx", "split295207-ref.mscx", 5) ; }
```
Comment (Jojo-Schmitz) [inline mtest/libmscore/split/tst_split.cpp]: to keep numerical oder better move that test down

Comment (njvdberg) [reply mtest/libmscore/split/tst_split.cpp]: Done

**Comment by Jojo-Schmitz** [comment]
Really (also) about https://musescore.org/en/node/287998, not (just) https://musescore.org/en/node/295207 ?

**Comment by njvdberg** [comment]
@Jojo-Schmitz :
> Really (also) about https://musescore.org/en/node/287998, not (just) https://musescore.org/en/node/295207 ?

Only https://musescore.org/en/node/295207, not https://musescore.org/en/node/287998. But I didn't add that one, did I? :-)

**Comment by Jojo-Schmitz** [comment]
You did, in the initial comment to this PR:

Resolves: https://musescore.org/en/node/287998

**Comment by Jojo-Schmitz** [comment]
> Resolves: https://musescore.org/en/node/287998

You did, in the initial comment to this PR

**Comment by njvdberg** [comment]
Hm, i can't find back the very original commit but now it says [#295207] only which is correct.

**Comment by Jojo-Schmitz** [comment]
the initial comment to this PR, not the commit title or body

**Comment by njvdberg** [comment]
Sorry, I overlooked that. It's corrected.

**Pull request merged**